src/components/dialog-test.js

```diff
@@ -411,6 +411,86 @@ describes.realWin('Dialog', (env) => {
       expect(styleDuringInit).to.equal('display: none !important;');
     });
 
+    describe('fade background', () => {
+      it('should not register click event by default', async () => {
+        const old = Dialog.prototype.onGrayPaneClick_;
+        let wasClicked = false;
+        Dialog.prototype.onGrayPaneClick_ = function () {
+          wasClicked = true;
+        };
+
+        dialog = new Dialog(globalDoc, {});
+        const el = dialog.graypane_.getElement();
+        const openedDialog = await dialog.open(NO_ANIMATE);
+        await openedDialog.openView(view);
+
+        el.click();
+
+        expect(wasClicked).to.be.false;
+
+        Dialog.prototype.onGrayPaneClick_ = old;
+      });
+
+      it('should register click event', async () => {
+        const old = Dialog.prototype.onGrayPaneClick_;
+        let wasClicked = false;
+        Dialog.prototype.onGrayPaneClick_ = function () {
+          wasClicked = true;
+        };
+
+        dialog = new Dialog(globalDoc, {}, {}, {isClosable: false});
+        const el = dialog.graypane_.getElement();
+        const openedDialog = await dialog.open(NO_ANIMATE);
+        await openedDialog.openView(view);
+
+        el.click();
+
+        expect(wasClicked).to.be.true;
+
+        Dialog.prototype.onGrayPaneClick_ = old;
+      });
+
+      it('should have a click event that returns false', async () => {
+        dialog = new Dialog(globalDoc, {});
+        expect(dialog.onGrayPaneClick_()).to.be.false;
+      });
+
+      it('respects not closable', async () => {
+        dialog = new Dialog(
+          globalDoc,
+          {},
+          {},
+          {isClosable: false, shouldDisableBodyScrolling: true}
+        );
+        const el = dialog.graypane_.getElement();
+        const openedDialog = await dialog.open(NO_ANIMATE);
+        await openedDialog.openView(view);
+        expect(doc.body).to.have.class('swg-disable-scroll');
+
+        el.click();
+
+        expect(doc.body).to.have.class('swg-disable-scroll');
```

Comment:
Checking the presence of the swg-disable-scroll class seems like a roundabout way to verify this behavior. Can you spy on close() instead?

Reply:
When I started adding spies it broke event registration and stopped calling the onclick function.  I'm not saying it can't be done, just that everything I tried was broken until I turned off the spies.

Reply:
Can you add comments explaining that the swg-disable-scroll class is indicating whether the dialog is closed?